src/dataflow/src/render/reduce.rs

```diff
@@ -1683,11 +1768,63 @@ where
                             (AggregateFunc::SumInt64, AccumInner::SimpleNumber { accum, .. }) => {
                                 Datum::from(*accum)
                             }
-                            (AggregateFunc::SumFloat32, AccumInner::SimpleNumber { accum, .. }) => {
-                                Datum::Float32((((*accum as f64) / float_scale) as f32).into())
+                            (
+                                AggregateFunc::SumFloat32,
+                                AccumInner::Float {
+                                    accum,
+                                    pos_infs,
+                                    neg_infs,
+                                    nans,
+                                    non_nulls,
+                                },
+                            ) => {
+                                let inf_accum = accum.is_infinite();
+                                let neg_accum = accum.is_sign_negative();
+                                let pos_inf = *pos_infs > 0 || (inf_accum && !neg_accum);
+                                let neg_inf = *neg_infs > 0 || (inf_accum && neg_accum);
+
+                                if *non_nulls == 0 {
```

Comment:
should we assert / debug assert that `accum == 0` in this case? I thought we had something like that before but I don't see it from a skim

Reply:
you're right, null aggregation is handled above. I fixed it here https://github.com/MaterializeInc/materialize/pull/8492/commits/650518d6a45b2553fb3fe6a203481d398cdc0860